### ISSUE TITLE
Replace PyQt5 with pyside2

### ISSUE THREAD

**Issue by mkolar** [pypeclub/OpenPype#1676]
PyQt5 license is more restrictive than PySide2, so if at all possible we should replace it. We're not currently providing pre-built binaries, but if we decide to do it, this might be a potential blocker on that front.

**Comment by iLLiCiTiT** [comment]
This require to have a folder in OpenPype build (and code) that is added to `sys.path` on OpenPype launch but is not added to `PYTHONPATH` (currently we're lucky that none of hosts is using PyQt5). @antirotor would that be possible?

**Comment by iLLiCiTiT** [comment]
When this is done, all we need is to replace PyQt5 with PySide2 in modules... (and maybe fix few smaller issues here and there)

**Comment by antirotor** [comment]
> This require to have a folder in OpenPype build (and code) that is added to `sys.path` on OpenPype launch but is not added to `PYTHONPATH` (currently we're lucky that none of hosts is using PyQt5). @antirotor would that be possible?
> 
> EDIT:
> That may be an issue in code right? As pyenv/poetry can install python modules only to one directory...

Please remind me - this is because we must not mix pyside2 with pyqt5 in the host, right?

**Comment by mkolar** [comment]
> Please remind me - this is because we must not mix pyside2 with pyqt5 in the host, right?

not entirely. that is doable.

The main issue is their licensing. PyQt5 is available under a GPL or commercial license, and PySide2 under a LGPL license.

Effectivelly meaning that you ran provide pre-built binaries with pyside, but not with PyQt5. Even though we are providing source code even if there are binaries. It would still be a lot simpler to work with various distribution situations with LGPL license. That's actually the whole reason for existence of PySide

**Comment by iLLiCiTiT** [comment]
> this is because we must not mix pyside2 with pyqt5 in the host, right?

It is because host may not use it's python modules as first but use what is in PYTHONPATH as first and since Qt bindings are build for specific version of python it will crash.

## 2 issues
- Maya will crash on start if you have different than maya's PySide2 in PYTHONPATH because will try to use it before it's own PySide2
- The same for Blender where we install PySide2 inside it's python but then we would pass PySide2 from OpenPype with PYTHONPATH so it will crash because of same reason

**Comment by iLLiCiTiT** [comment]
And since we have to pass our PYTHONPATH with our python packages we need to have 2 separated directories.

I think the best would be to have 2 venvs one for python modules that should not be passed to any host (Python modules, Qt binding,...) and other we want to pass...

**Comment by iLLiCiTiT** [comment]
Could we install `PySide2` and `shiboken2` using wheels into different directory? We will probably loose auto version control but I don't see other option...

**Comment by antirotor** [comment]
> Could we install `PySide2` and `shiboken2` using wheels into different directory? We will probably loose auto version control but I don't see other option...

yeah, me to. I would add function to lib, something like `get_qt_bindings()` that will then get it on demand from that different directory. `create_env` can install it via Poetry to `.venv` but then move it out to that other dir - thus we will retain management via Poetry.
